backend/src/main/java/org/sefglobal/core/academix/service/SubCategoryService.java

```diff
@@ -40,4 +40,19 @@ public SubCategoryService(SubCategoryRepository subCategoryRepository, CategoryR
         }
         return subCategoryRepository.getAllByCategory_Id(categoryId);
     }
+
+    /**
+     * Retrieves the subcategory with the requested id
+     * @param subCategoryId which is the id of the requested subCategory
+     * @return SubCategory object
+     * @throws ResourceNotFoundException if a subcategory with the requested id doesn't exist
+     */
+    public SubCategory getSubCategoryById(Long subCategoryId) throws ResourceNotFoundException {
```

Comment:
Return a `CustomSubCategory`

Reply:
Changed return type to CustomSubCategory